src/packet.rs

```diff
@@ -96,10 +115,13 @@ impl MspParser {
 			},
 
 			MspParserState::Header2 => {
-				if input == ('M' as u8) {
-					self.state = MspParserState::Direction;
-				} else {
-					self.reset();
+				self.state = match input as char {
```

Comment:
I think this is conceptually unclear. The `M` or `X` are not the Direction, but the Start of Frame marker. Don't mix them.

Reply:
I agree, they are not direction, they seem more like version indication `M` for msp v1 and `X` for msp v2. what do you mean don't mix them?

Reply:
You use two variants of `MspParserState` that begin with `Direction`. This is wrong IMO because the direction is encoded in exactly the same way in both MSP V1 and MSP V2. So semantically it is weird to put V1/V2 suffix to them, because this part of the message has no difference what so ever in both versions of MSP.

Reply:
Ok, got it